src/Riok.Mapperly/Descriptors/MappingBuilders/EnumToStringMappingBuilder.cs

```diff
@@ -26,11 +28,56 @@ public static class EnumToStringMappingBuilder
 
     private static EnumToStringMapping BuildEnumToStringMapping(MappingBuilderContext ctx)
     {
-        var ignoredSourceMembers = ctx.Configuration.Enum.IgnoredSourceMembers.ToHashSet(SymbolTypeEqualityComparer.FieldDefault);
-        var explicitMappings = BuildExplicitValueMappings(ctx);
+        var enumMemberMappings = BuildEnumMemberMappings(ctx);
+        var fallbackMapping = BuildFallbackMapping(ctx);
+
+        if (fallbackMapping.FallbackExpression is LiteralExpressionSyntax fallbackLiteral)
+        {
+            enumMemberMappings = enumMemberMappings.Where(IsNotEquivalentTo(fallbackLiteral));
+        }
+
+        return new EnumToStringMapping(ctx.Source, ctx.Target, enumMemberMappings, fallbackMapping);
+    }
+
+    private static IEnumerable<EnumMemberMapping> BuildEnumMemberMappings(MappingBuilderContext ctx)
+    {
+        var namingStrategy = ctx.Configuration.Enum.NamingStrategy;
 
+        var ignoredSourceMembers = ctx.Configuration.Enum.IgnoredSourceMembers.ToHashSet(SymbolTypeEqualityComparer.FieldDefault);
         EnumMappingDiagnosticReporter.AddUnmatchedSourceIgnoredMembers(ctx, ignoredSourceMembers);
-        return new EnumToStringMapping(ctx.Source, ctx.Target, ctx.SymbolAccessor.GetAllFields(ctx.Source), explicitMappings);
+
+        var fields = ctx.SymbolAccessor.GetFieldsExcept(ctx.Source, ignoredSourceMembers);
+        var explicitValueMappings = BuildExplicitValueMappings(ctx);
+        var customNameMappings = ctx.BuildCustomNameStrategyMappings(ctx.Source);
+
+        foreach (var field in fields)
+        {
+            // source.Value1
+            var sourceSyntax = MemberAccess(FullyQualifiedIdentifier(ctx.Source), field.Name);
+
+            ExpressionSyntax targetSyntax;
```

Comment:
Nitpick: i think the readability could be improved with early yield returns + continue statements.

Reply:
Good point, updated with early yield returns and continue statements for better readability.